File: demos/cross-platform-dashboard/artifacts/flutter/viewspec_view.dart
import 'package:flutter/material.dart';

class ViewSpecActionIntent {
  final String id;
  final String kind;
  final String targetRef;
  final List<String> payloadBindings;
  final Map<String, Object?> payload;

  const ViewSpecActionIntent({
    required this.id,
    required this.kind,
    required this.targetRef,
    required this.payloadBindings,
    required this.payload,
  });
}

typedef ViewSpecActionCallback = void Function(ViewSpecActionIntent intent);

class ViewSpecView extends StatefulWidget {
  final Map<String, Object?> data;
  final ViewSpecActionCallback? onAction;

  const ViewSpecView({super.key, this.data = const <String, Object?>{}, this.onAction});

  @override
  State<ViewSpecView> createState() => _ViewSpecViewState();
}

class _ViewSpecViewState extends State<ViewSpecView> {
  final Map<String, Object?> _inputValues = <String, Object?>{"include_mobile": true, "owner_email": "launch@viewspec.dev", "phase_filter": "Demos"};
  final Map<String, Object?> _compiledPayloadValues = <String, Object?>{"kpi_blockers_value": "1", "launch_status": "On track"};
  final Map<String, Object?> _visibilityRules = <String, Object?>{"show_mobile_note": <String, Object?>{"compareValue": "Mobile", "compareValueType": "string", "fallbackValue": "Demos", "fallbackValueType": "string", "id": "show_mobile_note", "operator": "EQUALS", "sourceKind": "input", "sourceRef": "phase_filter", "targetRef": "binding:mobile_recording_note"}};
  late final TextEditingController _owner_emailController;

  @override
  void initState() {
    super.initState();
    _owner_emailController = TextEditingController(text: '${_inputValues["owner_email"] ?? ""}');
  }

  @override
  void dispose() {
    _owner_emailController.dispose();
    super.dispose();
  }

  Map<String, Object?> _collectPayload(List<String> payloadBindings) {
    final payload = <String, Object?>{};
    for (final bindingId in payloadBindings) {
      if (_inputValues.containsKey(bindingId)) {
        payload[bindingId] = _inputValues[bindingId];
      } else if (_compiledPayloadValues.containsKey(bindingId)) {
        payload[bindingId] = _compiledPayloadValues[bindingId];
      }
    }
    return payload;
  }

  bool _isEmptyRuleValue(Object? value) {
    return value == null || value == '' || (value is List && value.isEmpty);
  }

  String _ruleValueType(Object? value) {
    if (value == null) return 'null';
    if (value is bool) return 'boolean';
    if (value is num) return 'number';
    if (value is String) return 'string';
    if (value is List) return 'array';
    return 'object';
  }

  bool _sameScalar(Object? left, Object? right) {
    final leftType = _ruleValueType(left);
    final rightType = _ruleValueType(right);
    if (!const <String>{'null', 'string', 'number', 'boolean'}.contains(leftType) || leftType != rightType) return false;
    return left == right;
  }

  Object? _ruleSourceValue(Map<String, Object?> rule, Map<String, Object?> inputValues, Map<String, Object?> data) {
    final sourceRef = rule['sourceRef'] as String? ?? '';
    if (rule['sourceKind'] == 'input') {
      return inputValues.containsKey(sourceRef) ? inputValues[sourceRef] : rule['fallbackValue'];
    }
    return data.containsKey(sourceRef) ? data[sourceRef] : rule['fallbackValue'];
  }

  bool _evaluateRule(String ruleId, Map<String, Object?> inputValues, Map<String, Object?> data) {
    final rawRule = _visibilityRules[ruleId];
    if (rawRule is! Map) return true;
    final rule = rawRule.cast<String, Object?>();
    final value = _ruleSourceValue(rule, inputValues, data);
    final operator = rule['operator'];
    if (operator == 'IS_EMPTY') return _isEmptyRuleValue(value);
    if (operator == 'NOT_EMPTY') return !_isEmptyRuleValue(value);
    if (operator == 'IS_TRUE') return value == true;
    if (operator == 'IS_FALSE') return value == false;
    if (operator == 'EQUALS') return _sameScalar(value, rule['compareValue']);
    if (operator == 'NOT_EQUALS') return !_sameScalar(value, rule['compareValue']);
    return true;
  }

  String? _selectValue(String inputId, List<String> options) {
    final value = _inputValues[inputId];
    if (value is String && options.contains(value)) return value;
    if (options.isNotEmpty) return options.first;
    return null;
  }

  void _dispatchAction(String id, String kind, String targetRef, List<String> payloadBindings) {
    widget.onAction?.call(
      ViewSpecActionIntent(
        id: id,
        kind: kind,
        targetRef: targetRef,
        payloadBindings: payloadBindings,
        payload: _collectPayload(payloadBindings),
      ),
    );
  }

  @override
  Widget build(BuildContext context) {
    return SafeArea(
      key: const ValueKey("dom-region_root"),
      child: SingleChildScrollView(
        child: Padding(
          padding: const EdgeInsets.all(24),
          child: Column(
            mainAxisSize: MainAxisSize.min,
            crossAxisAlignment: CrossAxisAlignment.stretch,
              children: <Widget>[
                Wrap(
                  key: const ValueKey("dom-region_header"),
                  spacing: 12,
                  runSpacing: 12,
                  children: <Widget>[
                    Text('${widget.data["launch_title"] ?? "Launch Operations Dashboard"}', key: const ValueKey("dom-binding_launch_title"), style: TextStyle(fontSize: 16, fontWeight: FontWeight.w700, color: const Color(0xFF1F2937))),
                    Container(
                      key: const ValueKey("dom-binding_launch_status"),
                      padding: const EdgeInsets.symmetric(horizontal: 12, vertical: 6),
                      decoration: BoxDecoration(color: const Color(0xFFF0FDFA), border: Border.all(color: const Color(0xFF99F6E4)), borderRadius: BorderRadius.circular(999)),
                      child: Text('${widget.data["launch_status"] ?? "On track"}', style: TextStyle(fontSize: 14, fontWeight: FontWeight.w700, color: const Color(0xFF0F766E))),
                    ),
                    Text('${widget.data["launch_summary"] ?? "One ViewSpec source proves web, React, iOS, and Android compilation."}', key: const ValueKey("dom-binding_launch_summary"), style: TextStyle(fontSize: 16, fontWeight: FontWeight.w400, color: const Color(0xFF1F2937))),
                  ]
                ),
                GridView.count(
                  key: const ValueKey("dom-region_kpis"),
                  shrinkWrap: true,
                  physics: const NeverScrollableScrollPhysics(),
                  crossAxisCount: 2,
                  crossAxisSpacing: 16,
                  mainAxisSpacing: 16,
                  children: <Widget>[
                    Column(
                      key: const ValueKey("dom-motif_launch_kpis"),
                      mainAxisSize: MainAxisSize.min,
                      crossAxisAlignment: CrossAxisAlignment.stretch,
                      children: <Widget>[
                        Container(
                          key: const ValueKey("dom-motif_launch_kpis_kpi_emitters"),
                          padding: const EdgeInsets.all(16),
                          decoration: BoxDecoration(color: const Color(0xFFFFFFFF), border: Border.all(color: const Color(0xFFE2E8F0)), borderRadius: BorderRadius.circular(12)),
                          child: Column(
                            mainAxisSize: MainAxisSize.min,
                            crossAxisAlignment: CrossAxisAlignment.stretch,
                              children: <Widget>[
                                Text('${widget.data["kpi_emitters_label"] ?? "Emitter targets"}', key: const ValueKey("dom-binding_kpi_emitters_label"), style: TextStyle(fontSize: 12, fontWeight: FontWeight.w800, color: const Color(0xFF64748B))),
                                Text('${widget.data["kpi_emitters_value"] ?? "4"}', key: const ValueKey("dom-binding_kpi_emitters_value"), style: TextStyle(fontSize: 28, fontWeight: FontWeight.w900, color: const Color(0xFF0F172A))),
                              ]
                          ),
                        ),
                        Container(
                          key: const ValueKey("dom-motif_launch_kpis_kpi_docs"),
                          padding: const EdgeInsets.all(16),
                          decoration: BoxDecoration(color: const Color(0xFFFFFFFF), border: Border.all(color: const Color(0xFFE2E8F0)), borderRadius: BorderRadius.circular(12)),
                          child: Column(
                            mainAxisSize: MainAxisSize.min,
                            crossAxisAlignment: CrossAxisAlignment.stretch,
                              children: <Widget>[
                                Text('${widget.data["kpi_docs_label"] ?? "Docs pages"}', key: const ValueKey("dom-binding_kpi_docs_label"), style: TextStyle(fontSize: 12, fontWeight: FontWeight.w700, color: const Color(0xFF64748B))),
                                Text('${widget.data["kpi_docs_value"] ?? "8"}', key: const ValueKey("dom-binding_kpi_docs_value"), style: TextStyle(fontSize: 28, fontWeight: FontWeight.w900, color: const Color(0xFF0F172A))),
                              ]
                          ),
                        ),
                        Container(
                          key: const ValueKey("dom-motif_launch_kpis_kpi_assets"),
                          padding: const EdgeInsets.all(16),
                          decoration: BoxDecoration(color: const Color(0xFFFFFFFF), border: Border.all(color: const Color(0xFFE2E8F0)), borderRadius: BorderRadius.circular(12)),
                          child: Column(
                            mainAxisSize: MainAxisSize.min,
                            crossAxisAlignment: CrossAxisAlignment.stretch,
                              children: <Widget>[
                                Text('${widget.data["kpi_assets_label"] ?? "Demo assets"}', key: const ValueKey("dom-binding_kpi_assets_label"), style: TextStyle(fontSize: 12, fontWeight: FontWeight.w700, color: const Color(0xFF64748B))),
                                Text('${widget.data["kpi_assets_value"] ?? "7"}', key: const ValueKey("dom-binding_kpi_assets_value"), style: TextStyle(fontSize: 28, fontWeight: FontWeight.w900, color: const Color(0xFF0F172A))),
                              ]
                          ),
                        ),
                        Container(
                          key: const ValueKey("dom-motif_launch_kpis_kpi_blockers"),
                          padding: const EdgeInsets.all(16),
                          decoration: BoxDecoration(color: const Color(0xFFFFFFFF), border: Border.all(color: const Color(0xFFE2E8F0)), borderRadius: BorderRadius.circular(12)),
                          child: Column(
                            mainAxisSize: MainAxisSize.min,
                            crossAxisAlignment: CrossAxisAlignment.stretch,
                              children: <Widget>[
                                Text('${widget.data["kpi_blockers_label"] ?? "Blockers"}', key: const ValueKey("dom-binding_kpi_blockers_label"), style: TextStyle(fontSize: 12, fontWeight: FontWeight.w700, color: const Color(0xFF64748B))),
                                Text('${widget.data["kpi_blockers_value"] ?? "1"}', key: const ValueKey("dom-binding_kpi_blockers_value"), style: TextStyle(fontSize: 28, fontWeight: FontWeight.w900, color: const Color(0xFF0F172A))),
                              ]
                          ),
                        ),
                      ]
                    ),
                  ]
                ),
                Column(
                  key: const ValueKey("dom-region_status"),
                  mainAxisSize: MainAxisSize.min,
                  crossAxisAlignment: CrossAxisAlignment.stretch,
                  children: <Widget>[
                    Column(
                      key: const ValueKey("dom-motif_launch_status_table"),
                      mainAxisSize: MainAxisSize.min,
                      crossAxisAlignment: CrossAxisAlignment.stretch,
                      children: <Widget>[
                        Wrap(
                          key: const ValueKey("dom-motif_launch_status_table_row_html"),
                          spacing: 12,
                          runSpacing: 12,
                          children: <Widget>[
                            Text('${widget.data["row_html_label"] ?? "HTML/Tailwind"}', key: const ValueKey("dom-binding_row_html_label"), style: TextStyle(fontSize: 12, fontWeight: FontWeight.w700, color: const Color(0xFF0F766E))),
                            Text('${widget.data["row_html_state"] ?? "Live link ready"}', key: const ValueKey("dom-binding_row_html_state"), style: TextStyle(fontSize: 28, fontWeight: FontWeight.w900, color: const Color(0xFF0F766E))),
                            Text('${widget.data["row_html_owner"] ?? "web"}', key: const ValueKey("dom-binding_row_html_owner"), style: TextStyle(fontSize: 28, fontWeight: FontWeight.w900, color: const Color(0xFF0F766E))),
                          ]
                        ),
                        Wrap(
                          key: const ValueKey("dom-motif_launch_status_table_row_react"),
                          spacing: 12,
                          runSpacing: 12,
                          children: <Widget>[
                            Text('${widget.data["row_react_label"] ?? "React TSX"}', key: const ValueKey("dom-binding_row_react_label"), style: TextStyle(fontSize: 12, fontWeight: FontWeight.w700, color: const Color(0xFF0F766E))),
                            Text('${widget.data["row_react_state"] ?? "Runtime page ready"}', key: const ValueKey("dom-binding_row_react_state"), style: TextStyle(fontSize: 28, fontWeight: FontWeight.w900, color: const Color(0xFF0F766E))),
                            Text('${widget.data["row_react_owner"] ?? "web"}', key: const ValueKey("dom-binding_row_react_owner"), style: TextStyle(fontSize: 28, fontWeight: FontWeight.w900, color: const Color(0xFF0F766E))),
                          ]
                        ),
                        Wrap(
                          key: const ValueKey("dom-motif_launch_status_table_row_swiftui"),
                          spacing: 12,
                          runSpacing: 12,
                          children: <Widget>[
                            Text('${widget.data["row_swiftui_label"] ?? "SwiftUI"}', key: const ValueKey("dom-binding_row_swiftui_label"), style: TextStyle(fontSize: 12, fontWeight: FontWeight.w700, color: const Color(0xFF64748B))),
                            Text('${widget.data["row_swiftui_state"] ?? "External simulator recording"}', key: const ValueKey("dom-binding_row_swiftui_state"), style: TextStyle(fontSize: 28, fontWeight: FontWeight.w900, color: const Color(0xFF0F172A))),
                            Text('${widget.data["row_swiftui_owner"] ?? "mobile"}', key: const ValueKey("dom-binding_row_swiftui_owner"), style: TextStyle(fontSize: 28, fontWeight: FontWeight.w900, color: const Color(0xFF0F172A))),
                          ]
                        ),
                        Wrap(
                          key: const ValueKey("dom-motif_launch_status_table_row_flutter"),
                          spacing: 12,
                          runSpacing: 12,
                          children: <Widget>[
                            Text('${widget.data["row_flutter_label"] ?? "Flutter"}', key: const ValueKey("dom-binding_row_flutter_label"), style: TextStyle(fontSize: 12, fontWeight: FontWeight.w700, color: const Color(0xFF64748B))),
                            Text('${widget.data["row_flutter_state"] ?? "External emulator recording"}', key: const ValueKey("dom-binding_row_flutter_state"), style: TextStyle(fontSize: 28, fontWeight: FontWeight.w900, color: const Color(0xFF0F172A))),
                            Text('${widget.data["row_flutter_owner"] ?? "mobile"}', key: const ValueKey("dom-binding_row_flutter_owner"), style: TextStyle(fontSize: 28, fontWeight: FontWeight.w900, color: const Color(0xFF0F172A))),
                          ]
                        ),
                      ]
                    ),
                  ]
                ),
                Column(
                  key: const ValueKey("dom-region_form"),
                  mainAxisSize: MainAxisSize.min,
                  crossAxisAlignment: CrossAxisAlignment.stretch,
                  children: <Widget>[
                    if (_evaluateRule("show_mobile_note", _inputValues, widget.data))
                      Text('${widget.data["mobile_recording_note"] ?? "Mobile runtime recording is handed off because this Windows workspace has no simulator tooling."}', key: const ValueKey("dom-binding_mobile_recording_note"), style: TextStyle(fontSize: 16, fontWeight: FontWeight.w400, color: const Color(0xFFB45309))),
                    DropdownButtonFormField<String>(
                      key: const ValueKey("dom-input_phase_filter"),
                      decoration: InputDecoration(labelText: "Phase filter"),
                      value: _selectValue("phase_filter", const <String>["Demos", "Docs", "Mobile", "Launch"]),
                      items: <DropdownMenuItem<String>>[
                        DropdownMenuItem<String>(value: "Demos", child: Text("Demos")),
                        DropdownMenuItem<String>(value: "Docs", child: Text("Docs")),
                        DropdownMenuItem<String>(value: "Mobile", child: Text("Mobile")),
                        DropdownMenuItem<String>(value: "Launch", child: Text("Launch")),
                      ],
                      onChanged: (value) => setState(() => _inputValues["phase_filter"] = value ?? ''),
                    ),
                    TextField(
                      key: const ValueKey("dom-input_owner_email"),
                      controller: _owner_emailController,
                      decoration: InputDecoration(labelText: "Owner email"),
                      onChanged: (value) => setState(() => _inputValues["owner_email"] = value),
                    ),
                    SwitchListTile(
                      key: const ValueKey("dom-input_include_mobile"),
                      title: Text("Include mobile handoff"),
                      value: (_inputValues["include_mobile"] as bool?) ?? false,
                      onChanged: (value) => setState(() => _inputValues["include_mobile"] = value),
                    ),
                    ElevatedButton(
                      key: const ValueKey("dom-action_save_launch_review"),
                      onPressed: () => _dispatchAction("save_launch_review", "submit", "/launch-review", const <String>["phase_filter", "owner_email", "include_mobile", "launch_status", "kpi_blockers_value"]),
                      child: Text("Save launch review"),
                    ),
                  ]
                ),
              ]
          ),
        ),
      ),
    );
  }
}

File: demos/cross-platform-dashboard/artifacts/swiftui/ViewSpecView.swift
import SwiftUI

public struct ViewSpecActionIntent {
    public let id: String
    public let kind: String
    public let targetRef: String
    public let payloadBindings: [String]
    public let payload: [String: Any]
}

public struct ViewSpecView: View {
    public var data: [String: Any]
    public var onAction: ((ViewSpecActionIntent) -> Void)?
    @State private var inputValues: [String: Any] = ["include_mobile": true, "owner_email": "launch@viewspec.dev", "phase_filter": "Demos"]
    private let compiledPayloadValues: [String: Any] = ["kpi_blockers_value": "1", "launch_status": "On track"]
    private let visibilityRules: [String: [String: Any]] = ["show_mobile_note": ["compareValue": "Mobile", "compareValueType": "string", "fallbackValue": "Demos", "fallbackValueType": "string", "id": "show_mobile_note", "operator": "EQUALS", "sourceKind": "input", "sourceRef": "phase_filter", "targetRef": "binding:mobile_recording_note"]]

    public init(data: [String: Any] = [:], onAction: ((ViewSpecActionIntent) -> Void)? = nil) {
        self.data = data
        self.onAction = onAction
    }

    private func unwrapOptionalValue(_ value: Any) -> Any? {
        let mirror = Mirror(reflecting: value)
        guard mirror.displayStyle == .optional else { return value }
        return mirror.children.first?.value
    }

    private func textValue(_ key: String, fallback: String) -> String {
        guard let rawValue = data[key], let value = unwrapOptionalValue(rawValue) else { return fallback }
        return String(describing: value)
    }

    private func collectPayload(_ payloadBindings: [String]) -> [String: Any] {
        var payload: [String: Any] = [:]
        for bindingId in payloadBindings {
            if let value = inputValues[bindingId] {
                payload[bindingId] = value
            } else if let value = compiledPayloadValues[bindingId] {
                payload[bindingId] = value
            }
        }
        return payload
    }

    private func isEmptyRuleValue(_ value: Any?) -> Bool {
        guard let value = value else { return true }
        if let text = value as? String { return text.isEmpty }
        if let list = value as? [Any] { return list.isEmpty }
        return false
    }

    private func numericRuleValue(_ value: Any?) -> Double? {
        if let value = value as? Int { return Double(value) }
        if let value = value as? Double { return value }
        if let value = value as? Float { return Double(value) }
        return nil
    }

    private func ruleValueType(_ value: Any?) -> String {
        guard let value = value else { return "null" }
        if value is Bool { return "boolean" }
        if numericRuleValue(value) != nil { return "number" }
        if value is String { return "string" }
        if value is [Any] { return "array" }
        return "object"
    }

    private func sameScalar(_ left: Any?, _ right: Any?) -> Bool {
        let leftType = ruleValueType(left)
        let rightType = ruleValueType(right)
        guard ["null", "string", "number", "boolean"].contains(leftType), leftType == rightType else { return false }
        if leftType == "null" { return true }
        if leftType == "number" { return numericRuleValue(left) == numericRuleValue(right) }
        if leftType == "boolean" { return (left as? Bool) == (right as? Bool) }
        return (left as? String) == (right as? String)
    }

    private func ruleValue(_ rule: [String: Any], key: String) -> Any? {
        if (rule["\(key)Type"] as? String) == "null" { return nil }
        return rule[key]
    }

    private func ruleSourceValue(_ rule: [String: Any], inputValues: [String: Any], data: [String: Any]) -> Any? {
        let sourceRef = rule["sourceRef"] as? String ?? ""
        if (rule["sourceKind"] as? String) == "input" {
            if let value = inputValues[sourceRef] { return unwrapOptionalValue(value) ?? ruleValue(rule, key: "fallbackValue") }
            return ruleValue(rule, key: "fallbackValue")
        }
        if let value = data[sourceRef] { return unwrapOptionalValue(value) ?? ruleValue(rule, key: "fallbackValue") }
        return ruleValue(rule, key: "fallbackValue")
    }

    private func evaluateRule(_ ruleId: String, inputValues: [String: Any], data: [String: Any]) -> Bool {
        guard let rule = visibilityRules[ruleId] else { return true }
        let value = ruleSourceValue(rule, inputValues: inputValues, data: data)
        let op = rule["operator"] as? String ?? ""
        if op == "IS_EMPTY" { return isEmptyRuleValue(value) }
        if op == "NOT_EMPTY" { return !isEmptyRuleValue(value) }
        if op == "IS_TRUE" { return (value as? Bool) == true }
        if op == "IS_FALSE" { return (value as? Bool) == false }
        if op == "EQUALS" { return sameScalar(value, ruleValue(rule, key: "compareValue")) }
        if op == "NOT_EQUALS" { return !sameScalar(value, ruleValue(rule, key: "compareValue")) }
        return true
    }

    private func dispatchAction(id: String, kind: String, targetRef: String, payloadBindings: [String]) {
        onAction?(ViewSpecActionIntent(
            id: id,
            kind: kind,
            targetRef: targetRef,
            payloadBindings: payloadBindings,
            payload: collectPayload(payloadBindings)
        ))
    }

    public var body: some View {
        ScrollView {
            VStack(alignment: .leading, spacing: 8.32) {
                VStack(alignment: .leading, spacing: 12) {
                    Text(textValue("launch_title", fallback: "Launch Operations Dashboard"))
                    .font(.system(size: 16, weight: .bold))
                    .foregroundColor(Color(red: 0.122, green: 0.161, blue: 0.216))
                    .opacity(0.92)
                    .accessibilityIdentifier("dom-binding_launch_title")
                    Text(textValue("launch_status", fallback: "On track"))
                    .font(.system(size: 14, weight: .semibold))
                    .foregroundColor(Color(red: 0.059, green: 0.463, blue: 0.431))
                    .opacity(0.92)
                    .accessibilityIdentifier("dom-binding_launch_status")
                    .padding(.horizontal, 12)
                    .padding(.vertical, 6)
                    .background(Color(red: 0.941, green: 0.992, blue: 0.980))
                    .cornerRadius(999)
                    Text(textValue("launch_summary", fallback: "One ViewSpec source proves web, React, iOS, and Android compilation."))
                    .font(.system(size: 16, weight: .regular))
                    .foregroundColor(Color(red: 0.122, green: 0.161, blue: 0.216))
                    .opacity(0.92)
                    .accessibilityIdentifier("dom-binding_launch_summary")
                }
                .accessibilityIdentifier("dom-region_header")
                LazyVGrid(columns: Array(repeating: GridItem(.flexible()), count: 2), spacing: 16) {
                    VStack(alignment: .leading, spacing: 12) {
                        VStack(alignment: .leading, spacing: 12) {
                            Text(textValue("kpi_emitters_label", fallback: "Emitter targets"))
                            .font(.system(size: 12, weight: .heavy))
                            .foregroundColor(Color(red: 0.122, green: 0.161, blue: 0.216))
                            .accessibilityIdentifier("dom-binding_kpi_emitters_label")
                            Text(textValue("kpi_emitters_value", fallback: "4"))
                            .font(.system(size: 28, weight: .black))
                            .foregroundColor(Color(red: 0.122, green: 0.161, blue: 0.216))
                            .opacity(0.92)
                            .accessibilityIdentifier("dom-binding_kpi_emitters_value")
                        }
                        .padding(16)
                        .background(Color.white)
                        .overlay(RoundedRectangle(cornerRadius: 12).stroke(Color(red: 0.886, green: 0.910, blue: 0.941)))
                        .cornerRadius(12)
                        .accessibilityIdentifier("dom-motif_launch_kpis_kpi_emitters")
                        VStack(alignment: .leading, spacing: 12) {
                            Text(textValue("kpi_docs_label", fallback: "Docs pages"))
                            .font(.system(size: 12, weight: .bold))
                            .foregroundColor(Color(red: 0.122, green: 0.161, blue: 0.216))
                            .opacity(0.92)
                            .accessibilityIdentifier("dom-binding_kpi_docs_label")
                            Text(textValue("kpi_docs_value", fallback: "8"))
                            .font(.system(size: 28, weight: .black))
                            .foregroundColor(Color(red: 0.122, green: 0.161, blue: 0.216))
                            .opacity(0.92)
                            .accessibilityIdentifier("dom-binding_kpi_docs_value")
                        }
                        .padding(16)
                        .background(Color.white)
                        .overlay(RoundedRectangle(cornerRadius: 12).stroke(Color(red: 0.886, green: 0.910, blue: 0.941)))
                        .cornerRadius(12)
                        .accessibilityIdentifier("dom-motif_launch_kpis_kpi_docs")
                        VStack(alignment: .leading, spacing: 12) {
                            Text(textValue("kpi_assets_label", fallback: "Demo assets"))
                            .font(.system(size: 12, weight: .bold))
                            .foregroundColor(Color(red: 0.122, green: 0.161, blue: 0.216))
                            .opacity(0.92)
                            .accessibilityIdentifier("dom-binding_kpi_assets_label")
                            Text(textValue("kpi_assets_value", fallback: "7"))
                            .font(.system(size: 28, weight: .black))
                            .foregroundColor(Color(red: 0.122, green: 0.161, blue: 0.216))
                            .opacity(0.92)
                            .accessibilityIdentifier("dom-binding_kpi_assets_value")
                        }
                        .padding(16)
                        .background(Color.white)
                        .overlay(RoundedRectangle(cornerRadius: 12).stroke(Color(red: 0.886, green: 0.910, blue: 0.941)))
                        .cornerRadius(12)
                        .accessibilityIdentifier("dom-motif_launch_kpis_kpi_assets")
                        VStack(alignment: .leading, spacing: 12) {
                            Text(textValue("kpi_blockers_label", fallback: "Blockers"))
                            .font(.system(size: 12, weight: .bold))
                            .foregroundColor(Color(red: 0.122, green: 0.161, blue: 0.216))
                            .opacity(0.92)
                            .accessibilityIdentifier("dom-binding_kpi_blockers_label")
                            Text(textValue("kpi_blockers_value", fallback: "1"))
                            .font(.system(size: 28, weight: .black))
                            .foregroundColor(Color(red: 0.122, green: 0.161, blue: 0.216))
                            .opacity(0.92)
                            .accessibilityIdentifier("dom-binding_kpi_blockers_value")
                        }
                        .padding(16)
                        .background(Color.white)
                        .overlay(RoundedRectangle(cornerRadius: 12).stroke(Color(red: 0.886, green: 0.910, blue: 0.941)))
                        .cornerRadius(12)
                        .accessibilityIdentifier("dom-motif_launch_kpis_kpi_blockers")
                    }
                    .accessibilityIdentifier("dom-motif_launch_kpis")
                }
                .accessibilityIdentifier("dom-region_kpis")
                VStack(alignment: .leading, spacing: 12) {
                    VStack(alignment: .leading, spacing: 12) {
                        VStack(alignment: .leading, spacing: 12) {
                            Text(textValue("row_html_label", fallback: "HTML/Tailwind"))
                            .font(.system(size: 12, weight: .bold))
                            .foregroundColor(Color(red: 0.059, green: 0.463, blue: 0.431))
                            .opacity(0.92)
                            .accessibilityIdentifier("dom-binding_row_html_label")
                            Text(textValue("row_html_state", fallback: "Live link ready"))
                            .font(.system(size: 28, weight: .black))
                            .foregroundColor(Color(red: 0.059, green: 0.463, blue: 0.431))
                            .opacity(0.92)
                            .accessibilityIdentifier("dom-binding_row_html_state")
                            Text(textValue("row_html_owner", fallback: "web"))
                            .font(.system(size: 28, weight: .black))
                            .foregroundColor(Color(red: 0.059, green: 0.463, blue: 0.431))
                            .opacity(0.92)
                            .accessibilityIdentifier("dom-binding_row_html_owner")
                        }
                        .accessibilityIdentifier("dom-motif_launch_status_table_row_html")
                        VStack(alignment: .leading, spacing: 12) {
                            Text(textValue("row_react_label", fallback: "React TSX"))
                            .font(.system(size: 12, weight: .bold))
                            .foregroundColor(Color(red: 0.059, green: 0.463, blue: 0.431))
                            .opacity(0.92)
                            .accessibilityIdentifier("dom-binding_row_react_label")
                            Text(textValue("row_react_state", fallback: "Runtime page ready"))
                            .font(.system(size: 28, weight: .black))
                            .foregroundColor(Color(red: 0.059, green: 0.463, blue: 0.431))
                            .opacity(0.92)
                            .accessibilityIdentifier("dom-binding_row_react_state")
                            Text(textValue("row_react_owner", fallback: "web"))
                            .font(.system(size: 28, weight: .black))
                            .foregroundColor(Color(red: 0.059, green: 0.463, blue: 0.431))
                            .opacity(0.92)
                            .accessibilityIdentifier("dom-binding_row_react_owner")
                        }
                        .accessibilityIdentifier("dom-motif_launch_status_table_row_react")
                        VStack(alignment: .leading, spacing: 12) {
                            Text(textValue("row_swiftui_label", fallback: "SwiftUI"))
                            .font(.system(size: 12, weight: .bold))
                            .foregroundColor(Color(red: 0.122, green: 0.161, blue: 0.216))
                            .opacity(0.92)
                            .accessibilityIdentifier("dom-binding_row_swiftui_label")
                            Text(textValue("row_swiftui_state", fallback: "External simulator recording"))
                            .font(.system(size: 28, weight: .black))
                            .foregroundColor(Color(red: 0.122, green: 0.161, blue: 0.216))
                            .opacity(0.92)
                            .accessibilityIdentifier("dom-binding_row_swiftui_state")
                            Text(textValue("row_swiftui_owner", fallback: "mobile"))
                            .font(.system(size: 28, weight: .black))
                            .foregroundColor(Color(red: 0.122, green: 0.161, blue: 0.216))
                            .opacity(0.92)
                            .accessibilityIdentifier("dom-binding_row_swiftui_owner")
                        }
                        .accessibilityIdentifier("dom-motif_launch_status_table_row_swiftui")
                        VStack(alignment: .leading, spacing: 12) {
                            Text(textValue("row_flutter_label", fallback: "Flutter"))
                            .font(.system(size: 12, weight: .bold))
                            .foregroundColor(Color(red: 0.122, green: 0.161, blue: 0.216))
                            .opacity(0.92)
                            .accessibilityIdentifier("dom-binding_row_flutter_label")
                            Text(textValue("row_flutter_state", fallback: "External emulator recording"))
                            .font(.system(size: 28, weight: .black))
                            .foregroundColor(Color(red: 0.122, green: 0.161, blue: 0.216))
                            .opacity(0.92)
                            .accessibilityIdentifier("dom-binding_row_flutter_state")
                            Text(textValue("row_flutter_owner", fallback: "mobile"))
                            .font(.system(size: 28, weight: .black))
                            .foregroundColor(Color(red: 0.122, green: 0.161, blue: 0.216))
                            .opacity(0.92)
                            .accessibilityIdentifier("dom-binding_row_flutter_owner")
                        }
                        .accessibilityIdentifier("dom-motif_launch_status_table_row_flutter")
                    }
                    .accessibilityIdentifier("dom-motif_launch_status_table")
                }
                .accessibilityIdentifier("dom-region_status")
                VStack(alignment: .leading, spacing: 16) {
                    if evaluateRule("show_mobile_note", inputValues: inputValues, data: data) {
                    Text(textValue("mobile_recording_note", fallback: "Mobile runtime recording is handed off because this Windows workspace has no simulator tooling."))
                    .font(.system(size: 16, weight: .regular))
                    .foregroundColor(Color(red: 0.706, green: 0.325, blue: 0.035))
                    .opacity(0.92)
                    .accessibilityIdentifier("dom-binding_mobile_recording_note")
                    }
                    Picker("Phase filter", selection: Binding<String>(
                        get: { inputValues["phase_filter"] as? String ?? "Demos" },
                        set: { inputValues["phase_filter"] = $0 }
                    )) {
                        Text("Demos").tag("Demos")
                        Text("Docs").tag("Docs")
                        Text("Mobile").tag("Mobile")
                        Text("Launch").tag("Launch")
                    }
                    .pickerStyle(.menu)
                    .accessibilityIdentifier("dom-input_phase_filter")
                    TextField("Owner email", text: Binding<String>(
                        get: { inputValues["owner_email"] as? String ?? "" },
                        set: { inputValues["owner_email"] = $0 }
                    ))
                    .textFieldStyle(.roundedBorder)
                    .accessibilityIdentifier("dom-input_owner_email")
                    Toggle("Include mobile handoff", isOn: Binding<Bool>(
                        get: { inputValues["include_mobile"] as? Bool ?? false },
                        set: { inputValues["include_mobile"] = $0 }
                    ))
                    .accessibilityIdentifier("dom-input_include_mobile")
                    Button(action: { dispatchAction(id: "save_launch_review", kind: "submit", targetRef: "/launch-review", payloadBindings: ["phase_filter", "owner_email", "include_mobile", "launch_status", "kpi_blockers_value"]) }) {
                        Text("Save launch review")
                    }
                    .buttonStyle(.borderedProminent)
                    .accessibilityIdentifier("dom-action_save_launch_review")
                }
                .accessibilityIdentifier("dom-region_form")
            }
            .padding(24)
        }
        .accessibilityIdentifier("dom-region_root")
    }
}
